railties/lib/rails/tasks/yarn.rake

```diff
@@ -10,7 +10,7 @@ namespace :yarn do
     end
 
     yarn_flags =
-      if `"#{Rails.root}/bin/yarn" --version`.start_with?("1")
+      if `#{RbConfig.ruby} "#{Rails.root}/bin/yarn" --version`.start_with?("1")
```

Comment:
Please add this prefix to the `yarn install` line as well. The install command itself is also currently broken on Windows.